Automattic/wp-calypso
# Checklist: Remove unused tasks and guided tours

#### Changes proposed in this Pull Request

We currently can potentially display just a few tasks in the site setup checklist (these are defined server-side in `Onboarding_Checklist\get_tasks()`):
- Set site title.
- Update site menu.
- Update homepage.
- Verify domain.
- Verify email.
- Launch site.
- Download mobile app.

This PRs removes the rest of tasks that are still present in the codebase in order to clean it up from dead code (and consequently improving the JS bundle size). It also removes their associated guided tours since they are no longer used.

The only exception to the above is the verify email guided tour. The task is still used but its guided tour wasn't showing up for some reason, so I decided to remove it rather than fixing it.

#### Testing instructions

- Create a new site.
- Go to My Home.
- Verify there are no changes to the site setup checklist when compared to production.

Part of https://github.com/Automattic/wp-calypso/issues/40646
Fixes https://github.com/Automattic/wp-calypso/issues/40872
